llama-index-core/llama_index/core/tools/types.py

```diff
@@ -70,6 +70,11 @@ def to_openai_function(self) -> Dict[str, Any]:
 
     def to_openai_tool(self) -> Dict[str, Any]:
         """To OpenAI tool."""
+        if len(self.description) > 1024:
```

Comment:
my one comment is that (sadly) this function is used in a few things outside the scope of openai (namely, inside the huggingface llm integration, and the mistral llm integration.

Should we have an optional flag to disable this check for these cases?

Reply:
Ohh interesting, I was wondering why it appears in core. I can definitely add a flag, should we default to skipping the check?

Reply:
Since its specific to openai, we should probably default to checking, and use the flag to disable

Reply:
Okay cool, should I update any of these other calls to set this flag? Will any break if we don't?

Reply:
yea lets update them to set the flag. They won't break per-say without it, but I'm not 100% sure if they have this limit